src/waves/waves.js

```diff
@@ -34,7 +34,9 @@ export class MdWaves {
     }
 
     this.attributeManager.addClasses(classes);
-    Waves.attach(this.element);
+    if (!(typeof window.require === 'function' && typeof window.require.specified === 'function' && window.require.specified('waves'))) {
```

Comment:
In my build, `window.require` is not defined but instead `window.requirejs`.. Is it a typo or does that actually work in your build?

Also: What do you think of a build condition instead? Since it's only required with AMD modules we could simply add it to AMD build (via a gulp "remove" task).